jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/Request.java

```diff
@@ -421,7 +422,16 @@ private void extractQueryParameters()
             try
             {
                 _queryParameters = new Fields(true);
-                UrlEncoded.decodeTo(query, _queryParameters::add, _queryEncoding);
+
+                if (StandardCharsets.UTF_8.equals(_queryEncoding) || _queryEncoding == null && UrlEncoded.ENCODING.equals(StandardCharsets.UTF_8))
+                {
+                    boolean allowBadUtf8 = getHttpChannel().getHttpConfiguration().getUriCompliance().allows(UriCompliance.Violation.BAD_UTF8_ENCODING);
+                    UrlEncoded.decodeUtf8To(query, 0, query.length(), _queryParameters::add, allowBadUtf8);
```

Comment:
The tracking of violations should occur here too.
If a violation occurred (the replacement character was used), then it should be added to the request attribute tracking those violations.

Reply:
done